spi/src/main/java/org/opensearch/jobscheduler/spi/utils/LockService.java

```diff
@@ -220,18 +215,16 @@ private void updateLock(final LockModel updateLock, ActionListener<LockModel> li
                 )
             );
         } catch (IOException e) {
-            logger.error("IOException occurred updating lock.", e);
-            listener.onResponse(null);
-        } catch (Exception e) {
-            logger.error(e);
+            logger.error("IOException occurred creating lock", e);
             listener.onFailure(e);
         }
     }
 
     private void createLock(final LockModel tempLock, ActionListener<LockModel> listener) {
-        try (ThreadContext.StoredContext ignore = client.threadPool().getThreadContext().stashContext()) {
+        try {
             final IndexRequest request = new IndexRequest(LOCK_INDEX_NAME).id(tempLock.getLockId())
                 .source(tempLock.toXContent(XContentFactory.jsonBuilder(), ToXContent.EMPTY_PARAMS))
+                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
```

Comment:
Not sure why this is necessary here just yet.

Reply:
This is needed to resolve a race condition discovered in [GetLockMultiNodeRestIT.testGetLockRestAPI](https://github.com/opensearch-project/job-scheduler/blob/main/src/test/java/org/opensearch/jobscheduler/multinode/GetLockMultiNodeRestIT.java#L50-L81).  This test will create and release a lock 10 times in a loop. Without adding this RefreshPolicy, when the security plugin is installed it can fail to find the lock the was just created. By setting this RefreshPolicy to IMMEDIATE, we can guarantee that this newly created document is available for search (including GET actions)